### PR TITLE
Fix Charging Moves, Reduction Berry Activation

### PR DESCRIPTION
**Pull request by kumakuma1**

### CODE REVIEW

File: data/battle_tests/moves/fly/mid_fight_neutral_damage.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            { ACTION_NONE, 0 },
+            { ACTION_NONE, 0 },
+            { ACTION_NONE, 0 },
+            { ACTION_NONE, 0 },
+            { ACTION_NONE, 0 },
+        } },
+    .expectations = {
+        { .expectationType = EXPECTATION_TYPE_MESSAGE, .expectationValue.message = "The opposing Mew used Fly!" },
+        { .expectationType = EXPECTATION_TYPE_MESSAGE, .expectationValue.message = "The opposing Mew flew up high!" },
+        { .expectationType = EXPECTATION_TYPE_MESSAGE, .expectationValue.message = "The opposing Mew became fully charged due to its Power Herb!" },
```
Comment (Aeliko) [inline data/battle_tests/moves/fly/mid_fight_neutral_damage.c:99]: Should we check the attack message isn't shown again?

File: data/battle_tests/moves/seismic_toss/effectiveness_berry.c [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            { ACTION_NONE, 0 },
+            { ACTION_NONE, 0 },
+            { ACTION_NONE, 0 },
+            { ACTION_NONE, 0 },
+            { ACTION_NONE, 0 },
+            { ACTION_NONE, 0 },
+        } },
+    .expectations = {
+        { .expectationType = EXPECTATION_TYPE_MESSAGE, .expectationValue.message = "Primeape used Seismic Toss!" },
+        { .expectationType = EXPECTATION_TYPE_MESSAGE_DOES_NOT_CONTAIN, .expectationValue.message = "It's super effective!" },
```
Comment (Aeliko) [inline data/battle_tests/moves/seismic_toss/effectiveness_berry.c:87]: We need to check that the berry doesn't activate and damage is not reduced in another test case, since we cannot have 2 negative test expectations

**Review by Aeliko — CHANGES_REQUESTED** [review]

File: data/battle_tests/moves/fly/mid_fight_neutral_damage.c (hunk 1 above)
Comment (kumakuma1) [reply data/battle_tests/moves/fly/mid_fight_neutral_damage.c:99]: added another message with flame orb proc to check it

File: data/battle_tests/moves/seismic_toss/effectiveness_berry.c (hunk 2 above)
Comment (kumakuma1) [reply data/battle_tests/moves/seismic_toss/effectiveness_berry.c:87]: added a second iteration with different expectation

File: src/individual/BattleController_BeforeMove.c [hunk 3]
```diff
@@ -2831,10 +2832,12 @@ BOOL BattleController_CheckTypeImmunity(struct BattleSystem *bsys, struct Battle
     }
     u32 flag = 0;
     int effectiveness = TYPE_MUL_NORMAL;
+    int target = ctx->moveTbl[ctx->current_move_index].target;
+    int power = ctx->moveTbl[ctx->current_move_index].power;
```
Comment (Aeliko) [inline src/individual/BattleController_BeforeMove.c:2836]: Can we check for move split instead of power?

**Review by Aeliko — APPROVED** [review]

File: src/individual/BattleController_BeforeMove.c (hunk 3 above)
Comment (kumakuma1) [reply src/individual/BattleController_BeforeMove.c:2836]: We ususally do check power for these things
